dm/dm/master/openapi_view.go

```diff
@@ -165,6 +181,31 @@ func (s *Server) DMAPIOfflineWorkerNode(c *gin.Context, workerName string) {
 	c.Status(http.StatusNoContent)
 }
 
+// DMAPIGetClusterInfo return cluster id of dm cluster url is: (GET /api/v1/cluster/info).
+func (s *Server) DMAPIGetClusterInfo(c *gin.Context) {
+	info, err := s.getClusterInfo(c.Request.Context())
+	if err != nil {
+		_ = c.Error(err)
+		return
+	}
+	c.IndentedJSON(http.StatusOK, info)
+}
+
+// DMAPIGetClusterInfo return cluster id of dm cluster url is: (PUT /api/v1/cluster/info).
+func (s *Server) DMAPIUpdateClusterInfo(c *gin.Context) {
```

Comment:
Just a question，it seems that only `updateClusterInfo` will put info to etcd,  so the topology will be accurate when use openapi, and it should be called by front-end?

Reply:
this info is mantained by tiup, and tiup can't write our etcd directly, so we add a openapi for it, of course users can write this info, but they should ensure the accuracy of this info themselves